src/components/GooglePlacesAutocomplete.jsx

```diff
@@ -0,0 +1,127 @@
+    const [suggestions, setSuggestions] = useState([]);
+    const [isLoading, setIsLoading] = useState(false);
+    const [showDropdown, setShowDropdown] = useState(false);
+
+    const dropdownRef = useRef(null);
+    const initialFetchDoneRef = useRef(false);
+
+    const handleSelect = useCallback(
+        (sug) => {
+            setInputValue(sug.description || sug.name || "");
+            setShowDropdown(false);
+            onSelect?.(sug);
+        },
+        [onSelect],
+    );
+
+    const handleInputChange = (e) => {
+        setInputValue(e.target.value);
+        setShowDropdown(true);
+        fetchSuggestions(e.target.value);
+    };
+
+    const fetchSuggestions = useMemo(
+        () =>
+            debounce(async (query, selectFirst = false) => {
+                if (!query.trim()) {
+                    setSuggestions([]);
+                    return;
+                }
+
+                setIsLoading(true);
+                try {
+                    const response = await fetch(`${urlConfig}/searchGooglePlaces?input=${encodeURIComponent(query)}`);
+                    const data = await response.json();
+                    const results = Array.isArray(data) ? data : data.predictions || [];
+                    setSuggestions(results);
+
+                    if (selectFirst && results.length > 0) {
+                        handleSelect(results[0]);
+                    }
+                } catch (error) {
+                    console.error("Error occurred:", error);
+                    setSuggestions([]);
+                } finally {
+                    setIsLoading(false);
+                }
+            }, 300),
+        [urlConfig, handleSelect],
+    );
+
+    useEffect(() => {
+        if (initialValue && !initialFetchDoneRef.current) {
+            initialFetchDoneRef.current = true;
+            fetchSuggestions(initialValue, true);
+        }
+    }, [initialValue, fetchSuggestions]);
+
+    /* eslint-disable no-undef */
+    useEffect(() => {
+        if (typeof document === "undefined") return;
+
+        const handleClickOutside = (e) => {
+            if (dropdownRef.current && !dropdownRef.current.contains(e.target)) {
+                setShowDropdown(false);
+            }
+        };
+
+        document.addEventListener("mousedown", handleClickOutside);
+
+        return () => {
+            document.removeEventListener("mousedown", handleClickOutside);
+        };
+    }, []);
+    /* eslint-enable no-undef */
+
+    return (
+        <div ref={dropdownRef} className="relative">
+            <Input type="text" value={inputValue} placeholder="Search place..." onChange={handleInputChange} />
+            {showDropdown && (
+                <div className="absolute z-10 max-h-65 w-full overflow-y-auto rounded-md border border-gray bg-white shadow-lg">
+                    {isLoading && <div className="text-gray-500 px-3 py-2">Loading...</div>}
```

Comment:
This is not a class in our UI Kit. Fix this. Fix the AI mistakes in this file please.